File: src/main/java/de/swtm/fuhrparkmanagement/model/Car.java
package de.swtm.fuhrparkmanagement.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.validation.constraints.Min;
import lombok.Data;
import lombok.EqualsAndHashCode;

@EqualsAndHashCode(callSuper = true)
@Data
@Entity
public class Car extends TemporalEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String licensePlate;

    private String modelName;

    private FuelType fuelType;

    private String location;

    @Min(1)
    private int seats;

    @Min(1)
    private int range;

    private boolean available;
}

File: app/src/app/api/models/car.ts
/* tslint:disable */
/* eslint-disable */
export interface Car {
  available?: boolean;
  createdDate?: string;
  deletedDate?: string;
  fuelType: 'PETROL' | 'DIESEL' | 'ELECTRIC';
  id?: number;
  licensePlate: string;
  location: string;
  modelName: string;
  range: number;
  seats: number;
}
